pgscatalog.core/src/pgscatalog/core/lib/_normalise.py

```diff
@@ -121,101 +114,43 @@ def drop_hla(variants):
     logger.warning(f"{n_dropped} HLA alleles detected and dropped")
 
 
-def check_effect_weight(variants):
-    """Check that effect weights are valid floats. Effect weights are intentionally
-    left as strings during processing.
-
-    >>> variant = ScoreVariant(**{"effect_allele": "A", "effect_weight": 5, "accession": "test", "row_nr": 0})
-    >>> list(check_effect_weight([variant])) # doctest: +ELLIPSIS
-    [ScoreVariant(effect_allele='A',effect_weight=5,...)]
-
-    >>> variant = ScoreVariant(**{"effect_allele": "A", "effect_weight": "potato", "accession": "test", "row_nr": 0})
-    >>> list(check_effect_weight([variant])) # doctest: +ELLIPSIS
-    Traceback (most recent call last):
-    ...
-    ValueError
-    """
-    for variant in variants:
-        try:
-            float(variant.effect_weight)
-        except ValueError as e:
-            logger.critical(f"{variant} has bad effect weight")
-            raise ValueError from e
-        else:
-            yield variant
```

Comment:
Handled by the pydantic model but we should think about how we handle effect weights 😅